torch/csrc/jit/passes/onnx/fixup_onnx_controlflow.cpp

```diff
@@ -342,6 +342,29 @@ void ONNXFixupUninitializedOutput(Node* node) {
           graph, else_block, else_block_output, then_block_output);
       if_node->outputs()[i]->setType(else_block->outputs()[i]->type());
     }
+    auto then_tensor_type =
+        then_block->outputs().at(i)->type()->castRaw<TensorType>();
+    auto else_tensor_type =
+        else_block->outputs().at(i)->type()->castRaw<TensorType>();
+    if (then_tensor_type && else_tensor_type) {
+      auto then_shape = then_tensor_type->symbolic_sizes();
+      auto else_shape = else_tensor_type->symbolic_sizes();
+      std::vector<::c10::ShapeSymbol> dims;
+      if (then_shape.rank() && else_shape.rank()) {
```

Comment:
Is it possible that only one of them has rank() available? If this case exists, can we just use this rank information to set `if_node->output(i)`?